src/main/kotlin/com/sourcegraph/cody/chat/ChatSession.kt

```diff
@@ -236,7 +254,29 @@ private constructor(private val project: Project, newSessionId: CompletableFutur
     @RequiresEdt
     fun createNew(project: Project): AgentChatSession {
       val sessionId = createNewPanel(project) { it.server.chatNew() }
-      val chatSession = AgentChatSession(project, sessionId)
+      val chatSession = getSession(sessionId.get()) ?: AgentChatSession(project, sessionId)
+      synchronized(chatSessions) { chatSessions.add(chatSession) }
+      return chatSession
+    }
+
+    fun createFromState(project: Project, state: ChatState): AgentChatSession {
+      val sessionId = createNewPanel(project) { it.server.chatNew() }
+      val chatSession =
+          getSession(sessionId.get()) ?: AgentChatSession(project, sessionId, state.internalId!!)
+      for (message in state.messages) {
+        val parsed =
+            when (val speaker = message.speaker) {
+              MessageState.SpeakerState.HUMAN -> Speaker.HUMAN
+              MessageState.SpeakerState.ASSISTANT -> Speaker.ASSISTANT
+              else -> error("unrecognized speaker $speaker")
+            }
+        val chatMessage = ChatMessage(parsed, message.text, id = UUID.randomUUID())
```

Comment:
`id = UUID.randomUUID()` is a default, you do not need to provide it explicitly.

Reply:
Fixed, thanks.